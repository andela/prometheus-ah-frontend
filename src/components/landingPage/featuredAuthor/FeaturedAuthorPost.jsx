import React from 'react';
import PropTypes from 'prop-types';


const FeaturedAuthorPost = ({
  title, image, readTime
}) => (
  <div className="d-flex justify-content-between featured-author-post mb-3">
    <div className="author-post">
      <h3 className="post-title">{title}</h3>
      <p><small className="text-muted font-italic">{readTime}</small></p>
    </div>
    <img src={image} className="img-fluid post-image" alt="" />
  </div>
);

FeaturedAuthorPost.propTypes = {
  image: PropTypes.string,
  title: PropTypes.string,
  readTime: PropTypes.string,
};

export default FeaturedAuthorPost;
