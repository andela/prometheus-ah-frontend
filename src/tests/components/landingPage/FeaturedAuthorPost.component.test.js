import React from 'react';
import { shallow } from 'enzyme';
import FeaturedAuthorPost from '../../../components/landingPage/featuredAuthor/FeaturedAuthorPost';
import mockData from '../../__mocks__/mockData';

let wrapper;

describe('FeaturedAuthorPost component', () => {
  beforeEach(() => {
    const {
      title, image, readTime
    } = mockData.postProps;
    const props = {
      title, image, readTime
    };

    wrapper = shallow(<FeaturedAuthorPost {...props} />);
  });

  test('should render a snapshot of the FeaturedAuthorPost component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
