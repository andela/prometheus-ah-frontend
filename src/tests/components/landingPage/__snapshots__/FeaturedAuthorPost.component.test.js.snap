// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedAuthorPost component should render a snapshot of the FeaturedAuthorPost component 1`] = `
<div
  className="d-flex justify-content-between featured-author-post mb-3"
>
  <div
    className="author-post"
  >
    <h3
      className="post-title"
    >
      My Broken Mother, My Broken Heart
    </h3>
    <p>
      <small
        className="text-muted font-italic"
      >
        1 min read
      </small>
    </p>
  </div>
  <img
    alt=""
    className="img-fluid post-image"
    src="https://image.ibb.co/eLVQwL/kids.jpg"
  />
</div>
`;
